parcels/particleset/particlesetnodes.py

```diff
@@ -0,0 +1,1389 @@
+        if self.repeatdt:
+            if self.repeatdt <= 0:
+                raise('Repeatdt should be > 0')
+            if time[0] and not np.allclose(time, time[0]):
+                raise ('All Particle.time should be the same when repeatdt is not None')
+            self.repeatpclass = _pclass
+
+        self._collection = ParticleCollectionNodes(self._idgen, self._c_lib_register, _pclass, lon=lon, lat=lat, depth=depth, time=time, lonlatdepth_dtype=lonlatdepth_dtype, pid_orig=pid_orig, partitions=partitions, ngrid=ngrids, **kwargs)
+
+        self.repeatlon = None
+        self.repeatlat = None
+        self.repeatdepth = None
+        self.repeatkwargs = None
+        self.repeat_starttime = None
+        if self.repeatdt:
+            self.repeatlon = np.empty(len(self._collection), dtype=self._lonlatdepth_dtype)
+            self.repeatlat = np.empty(len(self._collection), dtype=self._lonlatdepth_dtype)
+            self.repeatdepth = np.empty(len(self._collection), dtype=self._lonlatdepth_dtype)
+            self.repeatkwargs = {}
+            for kwvar in kwargs.keys():
+                self.repeatkwargs[kwvar] = []
+
+            collect_time = [pdata.time for pdata in self._collection]
+            if len(time) > 0 and (time[0] is None or np.isnan(time[0])):
+                self.repeat_starttime = time[0]
+            else:
+                if collect_time and not np.allclose(collect_time, collect_time[0]):
+                    raise ValueError('All Particle.time should be the same when repeatdt is not None')
+                self.repeat_starttime = collect_time[0]
+
+            base_time = None
+            pidx = 0
+            for pdata in self._collection:
+                self.repeatlon[pidx] = pdata.lon
+                self.repeatlat[pidx] = pdata.lat
+                self.repeatdepth[pidx] = pdata.depth
+                for kwvar in kwargs:
+                    self.repeatkwargs[kwvar].append(getattr(pdata, kwvar))
+                pidx += 1
+
+        self.repeatpid = None
+        if self.repeatdt:
+            if MPI and self._collection.pu_indicators is not None:
+                mpi_comm = MPI.COMM_WORLD
+                mpi_rank = mpi_comm.Get_rank()
+                self.repeatpid = None if pid_orig is None else pid_orig[self._collection.pu_indicators == mpi_rank]
+        # ---- ---- ---- ---- ---- finalize repeat-release parameters ---- ---- ---- ---- ---- #
+        # ------------------------------------------------------------------------------------ #
+
+        self._kernel = None
+        self._kclass = KernelNodes
+
+    def __del__(self):
+        """
+        ParticleSet - Destructor
+        """
+        if self._collection is not None:  # collection needs to be deleted here specifically.
+            self._collection.clear()
+            del self._collection
+        self._collection = None
+        super(ParticleSetNodes, self).__del__()
```

Comment:
```suggestion
        try:
            self._collection = None
            super(ParticleSetNodes, self).__del__()        
        except:    
            pass
```
Putting a try-except clause around garbage-collector to avoid confusing error messages to users when unlinking is unsuccessful

Reply:
I have a more fail-safe fix for that locally, which I'll try to commit soon.

Reply:
OK, look forward to seeing that then